mico-core/src/main/java/io/github/ust/mico/core/service/GitHubCrawler.java

```diff
@@ -53,48 +61,115 @@ public GitHubCrawler(RestTemplate restTemplate, KubernetesNameNormalizer kuberne
                 .setGitReleaseInfoUrl(releaseInfoJson.get("url").textValue());
+            if (!StringUtils.isEmpty(dockerfilePath)) {
+                if (existsFileInGithubRepo(gitHubRepositoryApiUrl, dockerfilePath)) {
+                    micoService.setDockerfilePath(dockerfilePath);
+                } else {
+                    throw new IllegalArgumentException("The Dockerfile path must be a valid path relativ to the repository root");
+                }
+            }
+            return micoService;
+        } else {
+            throw new IllegalArgumentException("An error occurred while requesting information about the GitHub repository. Send two requests and got the status codes "
+                + responseBasicInfo.getStatusCode() + " and " + responseReleaseInfo.getStatusCode().is2xxSuccessful());
+        }
 
-        } catch (IOException e) {
-            log.error(e.getStackTrace().toString());
-            log.error("Getting exception '{}'", e.getMessage());
-            throw e;
+    }
+
+    /**
+     * Checks if a file in a GitHub repository exists
+     *
+     * @param uriBasicInfo   the URI to a specific release of a GitHub repository
+     * @param dockerfilePath the relative path to the Dockerfile. The path is relative to the root directory of the
+     *                       GitHub repository
+     * @return {@code true} if the Dockerfile exists in the in the specified repository. {@code false} if response status is HTTP NOT FOUND.
+     * @throws {@link HttpClientErrorException} in case of HTTP 4XX response from GitHub
+     * @throws {@link HttpStatusCodeException} in case of HTTP 5XX response from GitHub
+     */
+    private boolean existsFileInGithubRepo(String uriBasicInfo, String dockerfilePath) {
+        ResponseEntity<String> responseDockerFileInfo;
+        UriComponentsBuilder dockerFileUriBuilder = UriComponentsBuilder.fromHttpUrl(uriBasicInfo);
+        UriComponents dockerFileUriComponent = dockerFileUriBuilder.pathSegment(GITHUB_API_CONTENTS).pathSegment(dockerfilePath).build();
+        log.debug("Check if the Dockerfile exists at {}", dockerFileUriComponent.toString());
+        try {
+            responseDockerFileInfo = restTemplate.getForEntity(dockerFileUriComponent.toUri(), String.class);
+            HttpStatus responseStatus = responseDockerFileInfo.getStatusCode();
+            if (responseStatus.equals(HttpStatus.OK)) {
+                log.debug("The file {} exists", dockerfilePath);
+                return true;
+            }
+        } catch (HttpClientErrorException.NotFound e) {
+            log.info("The file{} does not exist", dockerfilePath);
+            return false;
         }
+        return false;
     }
 
-    public MicoService crawlGitHubRepoLatestRelease(String uri) throws IOException {
+    public MicoService crawlGitHubRepoLatestRelease(String uri,@Nullable String dockerfilePath) throws IOException {
```

Comment:
To be conistent, it should be `url` instead of `uri`. Also consider to use a more specific naming like `gitHubRepoUrl`. Also ensure to format the file (space is missing).

Reply:
Done